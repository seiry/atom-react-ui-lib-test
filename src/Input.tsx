import { atom, useAtomValue, useSetAtom } from 'jotai';
import { ChangeEventHandler, FC, useCallback, useEffect } from 'react';
const inputAtom = atom('');

export const Input: FC = () => {
  const setText = useSetAtom(inputAtom);
  const handleChange: ChangeEventHandler<HTMLInputElement> = useCallback(
    (e) => setText(e.target.value),
    [setText]
  );

  return <input type="text" onChange={handleChange} />;
};

export const Button: FC = () => {
  const text = useAtomValue(inputAtom);

  return <button>Click me {text}</button>;
};

interface CombinationProps {
  defaultText?: string;
}

export const Combination: FC<CombinationProps> = (props) => {
  const { defaultText } = props;

  useEffect(() => {
    return () => {};
  }, []);

  return (
    <>
      <h1>{defaultText}</h1>
      <Input />
      <Button />
    </>
  );
};
