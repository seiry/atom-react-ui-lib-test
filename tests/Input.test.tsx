import renderer from 'react-test-renderer';
import { Combination } from '../src';

it('renders correctly', () => {
  const tree = renderer
    .create(<Combination defaultText="I'm groot" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
