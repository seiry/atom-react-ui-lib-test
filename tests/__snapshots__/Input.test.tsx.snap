// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <h1>
    I'm groot
  </h1>,
  <input
    onChange={[Function]}
    type="text"
  />,
  <button>
    Click me 
  </button>,
]
`;
